### PR TITLE
ECKey.findRecoveryId

### PR DESCRIPTION
**Pull request by matthewleon**
addresses https://github.com/bitcoinj/bitcoinj/issues/1568

### CODE REVIEW

File: core/src/test/java/org/bitcoinj/core/ECKeyTest.java [hunk 1]
```diff
@@ -239,6 +239,19 @@ public void verifyMessage() throws Exception {
         assertEquals(expectedAddress, gotAddress);
     }
 
+    @Test
```
Comment (schildbach) [inline core/src/test/java/org/bitcoinj/core/ECKeyTest.java]: Thanks for the test. But shouldn't it test for a specific recovery id? Currently it just asserts that the recovery id is between 0 and 3.

Is there a test vector in a specification, or in a different implementation perhaps?

Comment (matthewleon) [reply core/src/test/java/org/bitcoinj/core/ECKeyTest.java]: It would certainly be ideal to test for a specific recovery ID. I'll see what I can dig up.

**Comment by matthewleon** [comment]
@schildbach I've added two new tests. One uses a vector I found on Stack Overflow, which, while not ideal, seems to be common practice in lieu of a set of "best practice" vector for sepc256k1. The second is a version of another test in the file that replaces a looped search for the recovery ID with one that uses the new method.

**Comment by schildbach** [comment]
Thanks! I just squashed and merged.

**Pull request closed without merging**